### ISSUE TITLE
get_memory_usage causes segfault on Mac OS X.

### ISSUE THREAD

**Issue by monien** [flintlib/flint#1909]
The function get_memory_usage defined in src/generic_files/profiler.c uses /proc/self/status which does not exist on Mac OS X. Reading from it causes a segfault.

**Comment by albinahlback** [comment]
Yeah, there are also problems with non-x86 systems using `profiler.h` (I suspect you are affected by this as well).

If you find a solution, please let us know!
